src/orquestra/sdk/_base/_driver/_client.py

```diff
@@ -580,11 +580,29 @@ def get_workflow_run_result(
 
         _handle_common_errors(resp)
 
-        # Bug with mypy and Pydantic:
-        #   Unions cannot be passed to parse_obj_as: pydantic/pydantic#1847
-        return pydantic.parse_obj_as(
-            WorkflowResult, resp.json()  # type: ignore[arg-type]
-        )
+        # To ensure the correct ordering of results, we serialize the results on CE as:
+        # {
+        #   "results": [
+        #       (JSONResult | PickleResult).json(),
+        #       (JSONResult | PickleResult).json(),
+        #       ...
+        #   ]
+        # } aka a ComputeEngineWorkflowResult.json()
+        # For older workflows, we respond with:
+        # (JSONResult | PickleResult).json()
+
+        json_response = resp.json()
+        try:
+            # Try an older response
+            # Bug with mypy and Pydantic:
+            #   Unions cannot be passed to parse_obj_as: pydantic/pydantic#1847
+            return pydantic.parse_obj_as(
+                WorkflowResult, json_response  # type: ignore[arg-type]
+            )
+        except pydantic.ValidationError:
+            # If we fail, try parsing each part of a list separately
+            result_list = ComputeEngineWorkflowResult.parse_obj(json_response)
```

Comment:
note to self: why is this a `_list`?

Reply:
It used to be a list, before being made an explicit type 🙈 
I will rename it